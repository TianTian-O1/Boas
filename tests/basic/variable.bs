x = 10
y = 20.5
name = "Boas"
is_valid = true
result = x + y * 2

print(x)
print(y)
print(name)
print(is_valid)
print(result)