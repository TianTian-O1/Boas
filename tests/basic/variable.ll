; ModuleID = 'boas_module'
source_filename = "boas_module"
target datalayout = "e-m:e-p270:32:32-p271:32:32-p272:64:64-i64:64-f80:128-n8:16:32:64-S128"
target triple = "x86_64-unknown-linux-gnu"

@0 = private unnamed_addr constant [6 x i8] c"%.6g\0A\00", align 1
@1 = private unnamed_addr constant [6 x i8] c"%.6g\0A\00", align 1
@2 = private unnamed_addr constant [6 x i8] c"Boas\0A\00", align 1
@3 = private unnamed_addr constant [6 x i8] c"true\0A\00", align 1
@4 = private unnamed_addr constant [6 x i8] c"%.6g\0A\00", align 1

declare ptr @createList(i64)

declare double @listGetItem(ptr, i64)

declare void @listAppend(ptr, double)

declare double @listPop(ptr)

declare i64 @listSize(ptr)

declare i32 @printf(ptr, ...)

define i32 @main() {
entry:
  %0 = call i32 (ptr, ...) @printf(ptr @0, double 1.000000e+01)
  %1 = call i32 (ptr, ...) @printf(ptr @1, double 2.050000e+01)
  %2 = call i32 (ptr, ...) @printf(ptr @2)
  %3 = call i32 (ptr, ...) @printf(ptr @3)
  %4 = call i32 (ptr, ...) @printf(ptr @4, double 5.100000e+01)
  ret i32 0
}
