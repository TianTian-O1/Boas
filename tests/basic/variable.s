	.text
	.file	"boas_module"
	.section	.rodata.cst8,"aM",@progbits,8
	.p2align	3, 0x0                          # -- Begin function main
.LCPI0_0:
	.quad	0x4024000000000000              # double 10
.LCPI0_1:
	.quad	0x4034800000000000              # double 20.5
.LCPI0_2:
	.quad	0x4049800000000000              # double 51
	.text
	.globl	main
	.p2align	4
	.type	main,@function
main:                                   # @main
	.cfi_startproc
# %bb.0:                                # %entry
	pushq	%rax
	.cfi_def_cfa_offset 16
	leaq	.L__unnamed_1(%rip), %rdi
	movsd	.LCPI0_0(%rip), %xmm0           # xmm0 = [1.0E+1,0.0E+0]
	movb	$1, %al
	callq	printf@PLT
	leaq	.L__unnamed_2(%rip), %rdi
	movsd	.LCPI0_1(%rip), %xmm0           # xmm0 = [2.05E+1,0.0E+0]
	movb	$1, %al
	callq	printf@PLT
	leaq	.L__unnamed_3(%rip), %rdi
	xorl	%eax, %eax
	callq	printf@PLT
	leaq	.L__unnamed_4(%rip), %rdi
	xorl	%eax, %eax
	callq	printf@PLT
	leaq	.L__unnamed_5(%rip), %rdi
	movsd	.LCPI0_2(%rip), %xmm0           # xmm0 = [5.1E+1,0.0E+0]
	movb	$1, %al
	callq	printf@PLT
	xorl	%eax, %eax
	popq	%rcx
	.cfi_def_cfa_offset 8
	retq
.Lfunc_end0:
	.size	main, .Lfunc_end0-main
	.cfi_endproc
                                        # -- End function
	.type	.L__unnamed_1,@object           # @0
	.section	.rodata.str1.1,"aMS",@progbits,1
.L__unnamed_1:
	.asciz	"%.6g\n"
	.size	.L__unnamed_1, 6

	.type	.L__unnamed_2,@object           # @1
.L__unnamed_2:
	.asciz	"%.6g\n"
	.size	.L__unnamed_2, 6

	.type	.L__unnamed_3,@object           # @2
.L__unnamed_3:
	.asciz	"Boas\n"
	.size	.L__unnamed_3, 6

	.type	.L__unnamed_4,@object           # @3
.L__unnamed_4:
	.asciz	"true\n"
	.size	.L__unnamed_4, 6

	.type	.L__unnamed_5,@object           # @4
.L__unnamed_5:
	.asciz	"%.6g\n"
	.size	.L__unnamed_5, 6

	.section	".note.GNU-stack","",@progbits
