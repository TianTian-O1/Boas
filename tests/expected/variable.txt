10.00
20.50
Boas
true
51.00